examples/bazel_env/.bazelrc

```diff
@@ -1,15 +1,59 @@
-# Required on windows
+###############################################################################
+## Bazel Configuration Flags
+##
+## `.bazelrc` is a Bazel configuration file.
+## https://bazel.build/docs/best-practices#bazelrc-file
+###############################################################################
+
+# https://bazel.build/reference/command-line-reference#flag--enable_platform_specific_config
 common --enable_platform_specific_config
-startup --windows_enable_symlinks
-build:windows --enable_runfiles
 
-build --experimental_enable_bzlmod
+# Enable the only currently supported report type
+# https://bazel.build/reference/command-line-reference#flag--combined_report
+coverage --combined_report=lcov
+
+# Avoid fully cached builds reporting no coverage and failing CI
+# https://bazel.build/reference/command-line-reference#flag--experimental_fetch_all_coverage_outputs
+coverage --experimental_fetch_all_coverage_outputs
+
+# Required for some of the tests
+# https://bazel.build/reference/command-line-reference#flag--experimental_cc_shared_library
+common --experimental_cc_shared_library
+
+###############################################################################
+## Incompatibility flags
+###############################################################################
+
+# https://github.com/bazelbuild/bazel/issues/8195
+build --incompatible_disallow_empty_glob=true
 
-# This isn't currently the defaut in Bazel, but we enable it to test we'll be ready if/when it flips.
-build --incompatible_disallow_empty_glob
+# https://github.com/bazelbuild/bazel/issues/12821
+build --nolegacy_external_runfiles
 
 # Required for cargo_build_script support before Bazel 7
 build --incompatible_merge_fixed_and_default_shell_env
 
-# Do not import the PATH etc. from the host environment
-common --incompatible_strict_action_env
+# When our minimum bazel version reaches >= 7.4.0, we should uncomment the
```

Comment:
Let's just bump it? Same rationale for bumping to 7.3.1 justifies us bumping it to 7.4.0 :)

Reply:
https://github.com/bazelbuild/rules_rust/pull/3140 bump is here!